docs/docs/guides/developer_guides/smart_contracts/profiling_transactions.md

```diff
@@ -0,0 +1,70 @@
+---
+title: Profiling Transactions
+sidebar_position: 5
+tags: [contracts, profiling]
+---
+
+# Profiling Transactions
+
+An Aztec transaction typically consists of a private and a public part. The private part is where the user executes contract logic within the PXE and generates a proof of execution, which is then sent to the sequencer.
+
+Since proof generation is an expensive operation that needs to be done on the client side, it is important to optimize the private contract logic. It is desirable to keep the gate count of circuits representing the private contract logic as low as possible.
+
+A private transaction can involve multiple function calls. It starts with an account `entrypoint()` which may call several private functions to execute the application logic, which in turn might call other functions. Moreover, every private function call has to go through a round of kernel circuits. Read more about the transaction lifecycle [here](../../../aztec/concepts/transactions.md).
+
+In this guide, we will look at how to profile the private execution of a transaction, allowing you to get the gate count of each private function within the transaction, including the kernel circuits.
+
+## Prerequisites
+
+- `aztec-nargo` installed (go to [Sandbox section](../../../reference/developer_references/sandbox_reference/sandbox-reference.md) for installation instructions)
+- `aztec-wallet` installed (installed as part of the Sandbox)
+- Aztec Sandbox running with **proving enabled** (go to [Sandbox PXE Proving](../local_env/sandbox_proving.md) for instructions)
+
+## Profiling using aztec-wallet
+
+The profiling tool is integrated into the `aztec-wallet`.
+
+In this example, we will profile a simple "private token transfer" transaction which uses the [transfer](https://github.com/AztecProtocol/aztec-packages/blob/master/noir-projects/noir-contracts/contracts/token_contract/src/main.nr#L269) method in the token contract.
+
+Let's deploy the necessary account and token contracts first:
+
+```bash
+# Deploy accounts
+aztec-wallet create-account -a owner
+aztec-wallet create-account -a user
+
+# Deploy a token contract and mint 100 tokens to the user
+aztec-wallet deploy token_contract@Token --args accounts:owner Test TST 18 -f owner -a token
```

Comment:
Should we add a note that readers need to clone the monorepo and compile the token contract with `aztec-nargo compile --package token_contract` in noir-projects/noir-contracts?

Reply:
Oh, right. I will add that note.

> This is a lot to ask readers to go through, any ideas on how we can make this faster/easier?

In practice, devs would be trying this with their own contracts in their nargo workspace. This mostly serves as an example right?
But if they want to follow along and use the same steps then they would need to clone the monorepo and deploy the token contract. I think it is ok given that considering its not a mandatory step for anyone to profile their contract.